components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -188,12 +191,16 @@ public static String getOrganizationId() {
      *
      * @return the username of the authenticated user.
      */
-    public static String getAuthenticatedUsername() throws OrganizationManagementException {
+    public static String getAuthenticatedUsername() {
 
         String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
         if (username == null) {
-            return organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
-                    getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            try {
+                username = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
+                        getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            } catch (OrganizationManagementException e) {
+                LOG.debug("Authenticated user's username could not be resolved.", e);
```

Comment:
what's the reason for this change?

Reply:
Previously when organization is created, a tenant is created behind the scene with the organization creator's ID is set to the tenant admin's username. Later it was changed to keep the organization creator's username as the tenant admin's username. 

The authenticated organization creator can resides in higher level organization, but accessing a lower level organization to create an organization. This should be a cross tenant flow and the privilege carbon context does not keep the authenticated user's name for these types of flows. That's why the username had to resolved. 
https://github.com/wso2-extensions/identity-organization-management-core/pull/10